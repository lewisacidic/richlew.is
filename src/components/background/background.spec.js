import React from 'react'
import { render } from 'react-testing-library'

import Background from './background'
import { ThemeProvider } from 'styled-components'

describe('Background', () => {
  it('renders correctly', () => {
    const { getByTestId } = render(<Background data-testid="test-background" />)
    expect(getByTestId('test-background')).toMatchSnapshot()
  })
  it('will take a color as a background', () => {
    const { getByTestId } = render(
      <Background background="red" data-testid="test-background" />
    )
    expect(getByTestId('test-background')).toHaveStyle('background: red')
  })
  it('will take a theme prop as a background', () => {
    const { getByTestId } = render(
      <ThemeProvider theme={{ backgrounds: { lightRed: 'pink' } }}>
        <Background background="lightRed" data-testid="test-background" />
      </ThemeProvider>
    )
    expect(getByTestId('test-background')).toHaveStyle('background: pink')
  })
})
