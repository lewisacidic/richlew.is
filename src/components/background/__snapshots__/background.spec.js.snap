// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background renders correctly 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  position: absolute;
}

<div
  class="c0"
  data-testid="test-background"
/>
`;
